### PR TITLE
Merkurbank PDF Extractor: add Account Transactions

### PR DESCRIPTION
**Pull request by ZfT2**
added some basic account statements: deposit, removal

### CODE REVIEW

File: name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/merkurprivatbank/MerkurPrivatBankPDFExtractorTest.java [hunk 1]
```diff
@@ -160,4 +169,51 @@ public void testWertpapierVerkauf01()
...
+
+        assertThat(errors, empty());
+        assertThat(countSecurities(results), is(0L));
+        assertThat(countBuySell(results), is(0L));
+        assertThat(countAccountTransactions(results), is(3L));
+        assertThat(results.size(), is(3));
+        new AssertImportActions().check(results, CurrencyUnit.EUR);
+
+        assertThat(results, hasItem(deposit(hasDate("2024-01-19"), hasAmount("EUR", 900.00), //
+                        hasSource("Kontoauszug01.txt"))));
```
Comment (Nirus2000) [inline name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/merkurprivatbank/MerkurPrivatBankPDFExtractorTest.java:190]: To keep the tests complete, please add "hasNote" to your tests.

Thx 👍🏻

**Review by Nirus2000 — COMMENTED** [review]
Hey @ZfT2 
check my comments... i think it looks good with a few easy modifications... 👍🏻 

Good Job...

File: name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/merkurprivatbank/MerkurPrivatBankPDFExtractorTest.java (hunk 1 above)
Comment (ZfT2) [reply name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/merkurprivatbank/MerkurPrivatBankPDFExtractorTest.java]: OK, added it.

**Comment by ZfT2** [comment]
Thanks for reviewing it :) I updated the Importer with your suggestions.

File: name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/MerkurPrivatBankPDFExtractor.java [hunk 2]
```diff
@@ -124,4 +152,25 @@ private <T extends Transaction<?>> void addFeesSectionsTransaction(T transaction
...
+    private Transaction<AccountTransaction> depositRemovalTransaction(DocumentType type, String regex)
+    {
+        return new Transaction<AccountTransaction>().subject(() -> {
+            AccountTransaction entry = new AccountTransaction();
+            entry.setType(AccountTransaction.Type.DEPOSIT);
+            return entry;
+        }).section("date", "amount", "type").match(regex).assign((t, v) -> {
+            if ("S".equals(v.get("type")))
+                t.setType(Type.REMOVAL);
+            Map<String, String> context = type.getCurrentContext();
```
Comment (Nirus2000) [inline name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/MerkurPrivatBankPDFExtractor.java:165]: Hello @ZfT2 
You dont need `Map<String, String> context = type.getCurrentContext();`

add follow like -> [here](https://github.com/portfolio-performance/portfolio/blob/master/name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/DkbPDFExtractor.java#L723:L738) <-

`.documentContext("currency", "year") //`

then you can simply use `v.get("currency")` or `v.get("year")` to retrieve the string, as it is already parsed in the document.

**Review by Nirus2000 — CHANGES_REQUESTED** [review]

**Comment by ZfT2** [comment]
ok, changed it :)

**Pull request merged**